File: src/main/native/cpp/Sensors/OverPigeon/OverPigeon.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "OvertureLib/Sensors/OverPigeon/OverPigeon.h"

#ifndef __FRC_ROBORIO__
#include "OvertureLib/Simulation/SimPigeonManager/SimPigeonManager.h"
#endif

OverPigeon::OverPigeon(int deviceId, std::string canbus) : ctre::phoenix6::hardware::Pigeon2(
		deviceId, canbus) {
#ifndef __FRC_ROBORIO__
	SimPigeonManager &simPigeonManager = SimPigeonManager::GetInstance();
	simPigeonManager.SetSimPigeon(this);
#endif

	isConnectedAlert.SetText(
			"Pigeon " + std::to_string(deviceId) + " is not connected");
}

/**
 * @brief Updates the alert for the Pigeon
 */
void OverPigeon::updateAlert() {
	if (IsConnected()) {
		isConnectedAlert.Set(false);
	} else {
		isConnectedAlert.Set(true);
	}
}

File: src/main/java/com/overture/lib/sensors/OverPigeon.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package com.overture.lib.sensors;

import com.ctre.phoenix6.CANBus;
import com.ctre.phoenix6.hardware.Pigeon2;
import edu.wpi.first.wpilibj.Alert;

/** A Pigeon 2 that raises a driver station alert when it stops responding. */
public class OverPigeon extends Pigeon2 {
  private final Alert isConnectedAlert =
      new Alert("Devices", "Pigeon is not connected", Alert.AlertType.kWarning);

  /**
   * Constructs an OverPigeon on the default CAN bus.
   *
   * @param deviceId the CAN id of the Pigeon
   */
  public OverPigeon(int deviceId) {
    this(deviceId, new CANBus(""));
  }

  /**
   * Constructs an OverPigeon.
   *
   * @param deviceId the CAN id of the Pigeon
   * @param canbus the CAN bus the Pigeon lives on
   */
  public OverPigeon(int deviceId, CANBus canbus) {
    super(deviceId, canbus);

    isConnectedAlert.setText("Pigeon " + deviceId + " is not connected");
  }

  /** Raises the disconnection alert when the device stops responding. */
  public void updateConnectionAlert() {
    isConnectedAlert.set(!isConnected());
  }
}
